build-tools/packages/build-cli/src/commands/merge/branches.ts

```diff
@@ -102,22 +116,20 @@ export default class MergeBranch extends BaseCommand<typeof MergeBranch> {
 			flags.remote ?? (await this.gitRepo.getRemote(context.originRemotePartialUrl));
 		this.verbose(`Remote is: ${this.remote}`);
 
-		// Check if a branch integration PR exists already, based on its **name**.
-		const prExists: boolean = await pullRequestExists(
-			flags.pat,
-			prTitle,
-			owner,
-			repo,
-			this.logger,
-		);
+		if (flags.checkPr) {
+			// Check if a branch integration PR exists already, based on its **name**.
+			const prData = await pullRequestExists(flags.pat, prTitle, owner, repo, this.logger);
+
+			if (prData.found) {
+				this.log(`Found open PR #${prData.number} at ${prData.url}`);
+				return;
```

Comment:
@sonalideshpandemsft You were totally right about `return` here. If you call `exit`, it ends up calling catch as it exits - so that's a major difference between return and exit commands.